src/Microsoft.NetCore.Analyzers/Core/Security/UseSecureSharedAccesSignature.cs

```diff
@@ -0,0 +1,199 @@
+            nameof(SystemSecurityCryptographyResources.UseSecureSharedAccesSignatureDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor DoNotUseAccountSASRule = CreateDiagnosticDescriptor(s_DoNotUseAccountSASMessage);
+
+        internal static DiagnosticDescriptor UseSharedAccessProtocolHttpsOnlyRule = CreateDiagnosticDescriptor(s_UseSharedAccessProtocolHttpsOnlyMessage);
+
+        internal static DiagnosticDescriptor UseContainerLevelAccessPolicyRule = CreateDiagnosticDescriptor(s_UseContainerLevelAccessPolicyMessage);
+
+        private static DiagnosticDescriptor CreateDiagnosticDescriptor(
+            LocalizableString message)
+        {
+            return new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+        }
+
+        internal static ImmutableArray<(string, string)> NamespaceAndpolicyIdentifierNamePairs = ImmutableArray.Create(
+                                                                                                    ("Blob", "groupPolicyIdentifier"),
+                                                                                                    ("File", "groupPolicyIdentifier"),
+                                                                                                    ("Queue", "accessPolicyIdentifier"),
+                                                                                                    ("Table", "accessPolicyIdentifier"));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+                                                                                        DoNotUseAccountSASRule,
+                                                                                        UseSharedAccessProtocolHttpsOnlyRule,
+                                                                                        UseContainerLevelAccessPolicyRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
```

Comment:
We should definitely check to make sure Azure types are referenced by the compilation.  DFA can be expensive.